From 4a3d079e1c4ac21390dc3c4245a07e51801f7864 Mon Sep 17 00:00:00 2001
From: Demon000 <demonsingur@gmail.com>
Date: Mon, 12 Oct 2020 10:57:48 +0300
Subject: [PATCH 1/3] Partially revert "media: ov5640: Fix timings setup code"

This commit seems to cause some kind of race, causing the changed to
not come into effect before capture starts.
---
 drivers/media/i2c/ov5640.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/ov5640.c b/drivers/media/i2c/ov5640.c
index d332c45a89aa..15e622f8748d 100644
--- a/drivers/media/i2c/ov5640.c
+++ b/drivers/media/i2c/ov5640.c
@@ -1122,7 +1122,7 @@ static int ov5640_load_regs(struct ov5640_dev *sensor,
 			usleep_range(1000 * delay_ms, 1000 * delay_ms + 100);
 	}
 
-	return ov5640_set_timings(sensor, mode);
+	return ret;
 }
 
 static int ov5640_set_autoexposure(struct ov5640_dev *sensor, bool on)
@@ -1857,6 +1857,10 @@ static int ov5640_set_mode(struct ov5640_dev *sensor)
 	if (auto_exp)
 		ov5640_set_autoexposure(sensor, true);
 
+	ret = ov5640_set_timings(sensor, mode);
+	if (ret < 0)
+		return ret;
+
 	ret = ov5640_set_binning(sensor, dn_mode != SCALING);
 	if (ret < 0)
 		return ret;
-- 
2.28.0

